src/cs-core/SolarSystem.cpp

```diff
@@ -89,6 +89,25 @@ scene::CelestialObserver const& SolarSystem::getObserver() const {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+void SolarSystem::fixObserverFrame(double lastWorkingSimulationTime) {
+  // We try getting the position of the observer relative to the origin of the Solar System. If this
+  // fails, something is wrong with our observer frame.
+  try {
+    mObserver.getRelativePosition(mTimeControl->pSimulationTime.get(),
+        scene::CelestialAnchor("Solar System Barycenter", "J2000"));
+  } catch (...) {
+    // In case of an error, we reset the observer. This can throw an error itself, but we cannot do
+    // anything if that happens.
+    try {
+      mObserver.changeOrigin("Solar System Barycenter", "J2000", lastWorkingSimulationTime);
+    } catch (std::exception const& e) {
+      logger().warn("Failed to reset the Observer SPICE frame: {}", e.what());
```

Comment:
If this happens, shouldn't that be an error or critical error, since the application state is not recoverable?

Reply:
Good point!